src/main/cpp/src/BloomFilterJni.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cudf_jni_apis.hpp"
+#include "dtype_utils.hpp"
+#include "jni_utils.hpp"
+
+#include "bloom_filter.hpp"
+#include "utilities.hpp"
+
+extern "C" {
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_BloomFilter_put(JNIEnv* env,
+                                                                         jclass,
+                                                                         jlong bloom_filter,
+                                                                         jlong bloom_filter_bytes,
+                                                                         long bloom_filter_bits,
+                                                                         jlong cv,
+                                                                         jint num_hashes)
+{
+  try {
+    cudf::jni::auto_set_device(env);
+
+    cudf::column_view input_column{*reinterpret_cast<cudf::column_view const*>(cv)};
```

Comment:
Oh no this is a copy construction.
```suggestion
    cudf::column_view input_column = reinterpret_cast<cudf::column_view const*>(cv);
    // use `*input_column`
```

Reply:
Copy constructor of a column_view isn't a big deal.  I can make it a `const&` though:

`cudf::column_view const& input_column = *reinterpret_cast<cudf::column_view const*>(cv);`